src/inc/Vibrato.h

```diff
@@ -0,0 +1,82 @@
+#if !defined(__Vibrato_hdr__)
+#define __Vibrato_hdr__
+
+#include "ErrorDef.h"
+#include "Lfo.h"
+#include "RingBuffer.h"
+
+/*! \brief process audio data with vibrato
+*/
+
+class CVibrato {
+public:
+    /*!list of parameters for vibrato*/
+    enum VibratoParam_t {
+        kParamModFreq,          //!< ModFrequency
+        kParamModWidth,         //!< ModWidth
+        kParamDelay,            //!< Basic Delay
+        kParamSampleRate,       //!< Sample Rate
+        
+        kNumFilterParams
+    };
+    
+    /*！open a new instance for vibrato
+    \param pCVibrato
+    \return Error_t
+    */
+    static Error_t create (CVibrato*& pCVibrato);
+    
+    /*! destroy vibrato instance
+    \param pCVibrato
+    \return Error_t
+     */
+    static Error_t destroy (CVibrato*& pCVibrato);
+    
+    /*! set basic values, allocate memory for ringbuffer
+     \param fDelayInS
+     \param fSampleRateInHz
+     \param fModWidthInS
+     \param fModFreqInHz
+     \param iNumChannels
+     \return Error_t
+     */
+    Error_t init (float fDelayInS, float fSampleRateInHz, float fModWidthInS, float fModFreqInHz, int iNumChannels);
+    
+    /*! reset instance to initial state
+     \return Error_t
+     */
+    Error_t reset ();
+    
+    /*! set values for parameters
+     \param eParam
+     \return Error_t
+     */
+    Error_t setParam (VibratoParam_t eParam) const;
```

Comment:
I see that `setParam` is defined for `CLfo`, but I didn't see the implementation for `CVibrato` anywhere. Is it missing, or did I just miss it? Also, is there a corresponding `getParam`?

Reply:
I re-wrote the code for multiple versions, and I just found some part of it I got lost in my repo.
As the setParam doesn't use in normal main process, it wasn't detected.

Reply:
I will push a new version with setParam function. But the CLfo::setParam is not being used anymore.

Reply:
Updated.